test/e2e/network_test.go

```diff
@@ -148,6 +153,33 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+// getDeploymentLogs ensures logs of multiple deployments
+// are collected via a pod labelSelector
+func (tc *testContext) getDeploymentLogs(deployments ...*appsv1.Deployment) {
+	for _, deployment := range deployments {
+		labelSelector := deployment.Name
+		tc.savePodLogs(labelSelector)
+	}
+}
+
+// getLogs gets the logs for a job or deployment pod specified by the pod labelSelector
+func (tc *testContext) getLogs(podLabelSelector string) (string, error) {
+	if strings.HasPrefix(podLabelSelector, "job-name=") {
```

Comment:
Is this differentiating between handling a pod vs a deployment? If so, adding a comment mentioning that could be helpful

Reply:
Yes, I'll add that in

Reply:
We shouldn't be differentiating between a pod and deployment in this manner especially given that we have simple deployments without multiple containers.